lessons/basics/documentation.md

```diff
@@ -124,6 +124,8 @@ iex>
 
 Notice how you can use markup within our documentation and the terminal will render it? Apart from really being cool and a novel addition to Elixir's vast ecosystem, it gets much more interesting when we look at ExDoc to generate HTML documentation on the fly.
 
+**Note:** the `@spec` annotation is used to static analysis of code and is described [Specifications and types](../advanced/typespec) lesson.
```

Comment:
Just some small wording changes, thoughts?

**Note:** the `@spec` annotation is used for static analysis of code.  Too learn more about it, check out the [Specifications and types](../advanced/typespec) lesson.